src/plots.py

```diff
@@ -395,6 +395,79 @@ def plot_donut(df, col_color):
         },
     )
 
+def plot_donut_complex(df, col_color_1, col_color_2):
+    st.vega_lite_chart(
+        data=df,
+        spec={
+            "layer": [{
+                **CONFIG_MAIN,
+                "mark": {"type": "arc", "innerRadius": 75, "radius": 150, **CONFIG_MARK},
+                "transform": [
+                    {
+                        "window": [{"op": "count", "as": "total"}],
+                        "frame": [None, None],
+                    },
+                    {
+                        "joinaggregate": [{"op": "count", "as": "groupcount"}],
+                        "groupby": [col_color_1],
+                    },
+                    {"calculate": "datum.groupcount/datum.total", "as": "share"},
+                ],
+                "encoding": {
+                    "theta": {
+                        "type": "quantitative",
+                        "title": "Count",
+                        "aggregate": "count",
+                    },
+                    "color": {
+                        "field": col_color_1,
+                        "type": "nominal",
+                        "title": col_color_1.capitalize(),
+                    },
+                    "order": {
+                        "field": "share",
+                        "type": "quantitative",
+                        "sort": "descending",
+                        "title": "Share [%]",
+                        "format": ".1%",
+                    },
+                },
+            }, 
+            {
+                "mark": {"type": "arc", "innerRadius": 125, "radius": 175, **CONFIG_MARK},
+                "transform": [
+                    {
+                        "window": [{"op": "count", "as": "total"}],
+                        "frame": [None, None],
+                    },
+                    {
+                        "joinaggregate": [{"op": "count", "as": "groupcount"}],
+                        "groupby": [col_color_2],
```

Comment:
Shouldn't we group by col_color_2 and col_color_1? If we refer to the example plot in #6, the outside ring is describing the group count of col_color_2 for each value of col_color_1. You might need to double-check the sorting as well, in order to align everything correctly. Feel free to ping me if you need help on this one

Reply:
Hi, @fdebrain, i have updated PR according your comments, I will wait for you review.